### PR TITLE
Make `asyncio.isfuture` a `TypeGuard`

### PR DESCRIPTION
**Pull request by sobolevn**
Docs https://docs.python.org/3/library/asyncio-future.html#asyncio.isfuture

### CODE REVIEW

File: stdlib/asyncio/base_futures.pyi [hunk 1]
```diff
@@ -11,7 +11,7 @@ _PENDING: Literal["PENDING"]  # undocumented
 _CANCELLED: Literal["CANCELLED"]  # undocumented
 _FINISHED: Literal["FINISHED"]  # undocumented
 
-def isfuture(obj: object) -> bool: ...
+def isfuture(obj: object) -> TypeGuard[futures.Future]: ...
```
Comment (JelleZijlstra) [inline stdlib/asyncio/base_futures.pyi:14]: At runtime this allows any object with a specific attribute:
```
    return (hasattr(obj.__class__, '_asyncio_future_blocking') and
            obj._asyncio_future_blocking is not None)
```
I don't think that's enough for a TypeGuard.

Comment (sobolevn) [reply stdlib/asyncio/base_futures.pyi:14]: The docs put it this way:

https://docs.python.org/3/library/asyncio-future.html#asyncio.isfuture
> a Future-like object with a _asyncio_future_blocking attribute.

https://github.com/python/cpython/blob/c27a33132be101e246ae2584f1826477357138d6/Lib/asyncio/futures.py#L60-L63
>     # - Its presence is a marker to declare that a class implements
>     #   the Future protocol (i.e. is intended to be duck-type compatible).
>     #   The value must also be not-None, to enable a subclass to declare
>     #   that it is not compatible by setting this to None.

In my opinion, this is enough to be a `TypeGuard`

Comment (asvetlov) [reply stdlib/asyncio/base_futures.pyi:14]: The doc regarding `_asyncio_future_blocking` is correct.

I have another question. `asyncio.Future` is a generic type.
Should the typeguard reflect it, annotating at least `Future[Any]`?

Comment (sobolevn) [reply stdlib/asyncio/base_futures.pyi]: Done! Thanks!

**Review by asvetlov — APPROVED** [review]

**Pull request merged**